### PR TITLE
[SPARK-33290][SQL][DOCS][FOLLOW-UP] Update SQL migration guide

### PR DESCRIPTION
**Pull request by sunchao**
### What changes were proposed in this pull request?

Update SQL migration guide for SPARK-33290

### Why are the changes needed?

Make the change better documented.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @gatorsmile and @cloud-fan

**Review by gatorsmile — APPROVED** [review]

**Comment by gatorsmile** [comment]
Thank you for adding it! LGTM

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -52,6 +52,8 @@ license: |
   
   - In Spark 3.1, the `schema_of_json` and `schema_of_csv` functions return the schema in the SQL format in which field names are quoted. In Spark 3.0, the function returns a catalog string without field quoting and in lower case. 
 
+  - In Spark 3.1, refreshing a table will trigger a uncache operation for all other caches that reference the table, even if the table itself is not cached. In Spark 3.0 the operation will only be triggered if the table itself is cached.
```
Comment (dongjoon-hyun) [inline docs/sql-migration-guide.md:55]: `a uncache` -> `an uncache`?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except one minor comment about a typo). Thanks.

**Comment by dongjoon-hyun** [comment]
Merged to master~

**Comment by sunchao** [comment]
Thanks!

**Pull request closed without merging**